activesupport/lib/active_support/testing/method_call_assertions.rb

```diff
@@ -28,7 +28,7 @@ def assert_called_with(object, method_name, args, returns: nil, &block)
 
           object.stub(method_name, mock, &block)
 
-          mock.verify
+          assert_mock(mock)
```

Comment:
I believe this is a "public" way of verifying a mock:
https://github.com/seattlerb/minitest/blob/13c48a03d84a2a87855a4de0c959f96800100357/lib/minitest/mock.rb#L184